src/fomo.jl

```diff
@@ -0,0 +1,399 @@
+export fomo, FomoSolver, FoSolver, R2, tr_step, r2_step
+
+abstract type AbstractFirstOrderSolver <: AbstractOptimizationSolver end
+
+abstract type AbstractFomoMethod end
+struct tr_step   <: AbstractFomoMethod end
+struct r2_step   <: AbstractFomoMethod end
+
+"""
+    fomo(nlp; kwargs...)
+    R2(nlp; kwargs...)
+
+A First-Order with MOmentum (FOMO) model-based method for unconstrained optimization. Supports quadratic regularization and trust region methods.
+
+For advanced usage, first define a `FomoSolver` to preallocate the memory used in the algorithm, and then call `solve!`:
+
+    solver = FomoSolver(nlp)
+    solve!(solver, nlp; kwargs...)
+
+**Quadratic Regularization (R2)**: if the user do not want to use momentum (`β` = 0), it is recommended to use the memory-optimized `R2` method.
```

Comment:
Doesn’t this also apply to TR steps? It’s not just R2, right?

Reply:
As it is, the solver can do either R2 or TR steps. I think that’s good. But I’m saying the momentum discussion applies to both R2 and TR.